src/main/row.ts

```diff
@@ -59,6 +64,116 @@ export function hasId(row: Row): boolean {
     return row.metadata.getValue() !== null;
 }
 
+/**
+ * Saves a given row's contents to the PropertiesService
+ * 
+ * @param row the row's contents to write
+ * @returns boolean representing whether the save operation was successful or not
+ */
+export function saveRow(row: Row): boolean {
+    if(!hasId(row)) {
+        throw Error(`Row does not have an id: ${toString(row)}`);
+    }
+
+    const rowId: string = getId(row);
+    const rowHash: string = toHexString(Utilities.computeDigest(Utilities.DigestAlgorithm.SHA_256, toString(row)));
+
+    return setDocumentProperty(rowId, rowHash);
+}
+
+/**
+ * Batched version of saveRow() which saves an array of row's contents
```

Comment:
Yea wasn't sure if we would use a batched call or would loop through and save individual ones so I exposed the functionality for both. Efficiency wise I think the batched may be a little better because only one called to PropertiesService is made. Multiple calls to the PropertiesService may be a little expensive

Reply:
i'm not 100% sure yet either; let's keep both since you already wrote it anyway